pydra/engine/submitter.py

```diff
@@ -57,9 +57,10 @@ def __exit__(self, type, value, traceback):
         task_futures = set()
         while not wf.done:
             remaining_tasks, tasks = await get_runnable_tasks(wf.graph, remaining_tasks)
-            if not tasks and not task_futures:
-                breakpoint()
-                raise Exception("Nothing queued or todo - something went wrong")
+            # dj: i actually think that this is ok if tasks is empty...
+            # if not tasks and not task_futures:
```

Comment:
tasks can be empty, but if there are also no existing `task_futures` then nothing is being done and we will be stuck in an infinite loop

Did you run into errors with this after the results fix? I didn't with my hasty attempts

Reply:
yes, this also causes error!

Reply:
ok, I see you point, but the if should look like `if not tasks and not remaining_tasks`

Reply:
I'm actually not sure if I understand the concept behind `task_futures`